include/benchmark/benchmark.h

```diff
@@ -417,6 +417,8 @@ class Benchmark;
 class BenchmarkImp;
 class BenchmarkFamilies;
 
+BENCHMARK_EXPORT std::map<std::string, std::string>* GetGlobalContext();
```

Comment:
this was very deliberately _not_ intended to be exposed as part of the public API.

Reply:
I see that - that's why it's put in the `internal` namespace. Effectively, this is as "private" as the `global_context` field we had previously. Other callers were still able to access the "hidden" field by forward declaring it, so I don't think adding a getter here makes it any worse.

Reply:
i had hoped it would be more difficult or more obvious that someone was  doing something they shouldn't be, but the `internal` namespace _should_ ideally give that signal too.

one thing i'm curious about in general: what build issues does this fix that aren't showing up on the CI bots and why aren't they showing up on the CI bots?

Reply:
Previously, when we were getting undefined symbol error for the global_context when it's not marked with the export attribute. But adding the attribute causes "duplicate symbol definition" because of all the forward decl. (On the other hand, we cannot remove the forward decls because the field is used by other files).

So it seemed adding a properly exported getter function here and removing all the forward decls is the cleanest way to fix this. 

I don't know why the CI bots don't report the error. It's possible that the bots only tests the scenario where STATIC is set (ie., the import/export attribute is no-op)?

Reply:
the solution is fine. i'm just baffled because we have windows-2019 and windows-2022, debug and release, static and shared on the CI bots.

what configuration did you see the build errors in?

Reply:
(i'll send you the sponge links)

Reply:
gotcha. for posterity: it's windows+bazel+clang (mostly) which we don't cover in CI.